nuget/spec/dependabot/nuget/version_spec.rb

```diff
@@ -71,189 +71,39 @@
   end
 
   describe "#<=>" do
-    subject { version <=> other_version }
-
-    context "compared to a Gem::Version" do
-      context "that is lower" do
-        let(:other_version) { Gem::Version.new("0.9.0") }
-        it { is_expected.to eq(1) }
-      end
-
-      context "that is equal" do
-        let(:other_version) { Gem::Version.new("1.0.0") }
-        it { is_expected.to eq(0) }
-
-        context "but our version has build information" do
-          let(:version_string) { "1.0.0+gc.1" }
-          it { is_expected.to eq(0) }
-        end
-      end
-
-      context "that is greater" do
-        let(:other_version) { Gem::Version.new("1.1.0") }
-        it { is_expected.to eq(-1) }
+    sorted_versions = [
+      "",
+      "0.9.0",
+      "1.0.0-alpha",
+      "1.0.0-alpha.1",
+      "1.0.0-alpha.beta",
+      "1.0.0-beta",
+      "1.0.0-beta.2",
+      "1.0.0-beta.11",
+      "1.0.0-beta.11.1",
+      "1.0.0-beta-extra-hyphens",
+      "1.0.0-rc.1",
+      "1.0.0",
+      "1.1.0"
+    ]
+    sorted_versions.combination(2).each do |lhs, rhs|
```

Comment:
This is a cool approach!

Reply:
A slightly more idiomatic and performant way to write this is:

```suggestion
    sorted_versions.each_slice(2) do |lhs, rhs|
```

Performance-wise this doesn't matter at all when n = 12, but it might read slightly more familiar to rubyists. Will leave it up to you wether or not to make the change though, I don't feel strongly about this

Reply:
It looks like `each_slice` does something different than `combination`, in irb:

```ruby
irb(main):001:0> [1,2,3].combination(2).to_a
=> [[1, 2], [1, 3], [2, 3]]
irb(main):002:0> [1,2,3].each_slice(2).to_a
=> [[1, 2], [3]]
```

In this case I think we want to stick with `combination` so we can test every version against the others to catch mistakes like the one I made with `zip` 🙃 .